TestCases/compliance-level-3/0072-feel-in/0072-feel-in.dmn

```diff
@@ -0,0 +1,2178 @@
+        <literalExpression>
+            <text>date and time("2018-12-08T10:30:05") in ? = date and time("2018-12-08T10:30:05")</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="dateTime_013_a" id="_dateTime_013_a">
+        <variable name="dateTime_013_a"/>
+        <literalExpression>
+            <text>date and time("2018-12-08T10:30:05") in ? != date and time("2018-12-08T10:30:05")</text>
+        </literalExpression>
+    </decision>
+-->
+
+    <decision name="dateTime_014" id="_dateTime_014">
+        <variable name="dateTime_014"/>
+        <literalExpression>
+            <text>date and time("2018-12-08T10:30:05") in (date and time("2018-12-08T10:30:04"), >=date and time("2018-12-08T10:30:05"))</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="dateTime_014_a" id="_dateTime_014_a">
+        <variable name="dateTime_014_a"/>
+        <literalExpression>
+            <text>date and time("2018-12-08T10:30:05") in (date and time("2018-12-08T10:30:04"), >=date and time("2018-12-08T10:30:06"))</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="list_001" id="_list_001">
+        <variable name="list_001"/>
+        <literalExpression>
+            <text>[1,2,3] in [[1,2,3,4], [1,2,3]]</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="list_001_a" id="_list_001_a">
+        <variable name="list_001_a"/>
+        <literalExpression>
+            <text>[1,2,3,5] in [[1,2,3,4], [1,2,3]]</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="list_011" id="_list_011">
+        <variable name="list_011"/>
+        <literalExpression>
+            <!-- looks like it should be an equals form left to right .. but the list [1,2,3] does not
+            equal 1, 2 or 3] values -->
+            <text>[1,2,3] in [1,2,3]</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="list_011_a" id="_list_011_a">
+        <variable name="list_011_a"/>
+        <literalExpression>
+            <text>[1,2,3] in [[1,2,3], [1,2,3,4]]</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="list_011_b" id="_list_011_b">
+        <variable name="list_011_b"/>
+        <literalExpression>
+            <text>[1,2,2] in [[1,2,3], [1,2,3,4]]</text>
+        </literalExpression>
+    </decision>
+
+<!--
+    <decision name="list_013" id="_list_013">
+        <variable name="list_013"/>
+        <literalExpression>
+            <text>[1,2,3] in ? = [1,2,3]</text>
+        </literalExpression>
+    </decision>
+
+    <decision name="list_013_a" id="_list_013_a">
+        <variable name="list_013_a"/>
+        <literalExpression>
+            <text>[1,2,3] in ? != [1,2,3]</text>
+        </literalExpression>
+    </decision>
+-->
+
+    <decision name="list_014" id="_list_014">
+        <variable name="list_014"/>
+        <literalExpression>
+            <text>[1,2,3] in ([1,2,3,4], [1,2,3])</text>
```

Comment:
Rule 51.d as tested for instance by number_014
We apply the element, a list `[1,2,3]`, to a list of positive unary tests { `=[1,2,3,4]` , `=[1,2,3]`} .
Results are`{false, true}`.
End result, `true` (as number_014 test).

Reply:
Thanks @tarilabs. This test and the one you mentioned below are there specifically to catch an ambiguity - or at least what I feel is one.  That ambiguity is when the target of the unary tests is a list and a list appears as a test in a unary tests.  It is an equality match?  Or, as if we were (say) testing `1` in `([1,2,3])`?  We do not compare `1` against `[1,2,3]` - we compare it against each element of the list.    

So, the test shows that a list like `[1,2,3]` tested against `([1,2,3])` is comparing `[1,2,3]` against `1`, then `2` then `3`.  None of which match.

Where this become a bit strange is that the 'syntactic sugar' section mentions has a number of different interpretations of `in`.  Maybe, when lists are the target of a unary test we should be testing each element (as shown in prev paragraph) and *also* an equality check.  Like ... run down through the syntactic sugar stuff and evaluate relevant ones.

The spec doesn't give any guidance here as far as I can tell, so I threw in a test to see if my assumption was correct. 

Does that make any sense?

Reply:
@StrayAlien understood, and what @tarilabs described is the intent of the spec. The following expression is not ambiguous, because the comma in "( <value> ,  <value> )" disambiguates it:

> [1,2,3] in ([1,2,3,4], [1,2,3])

The result is described by @tarilabs .

Having said that, you raise a good point I didn't think before. The following *IS* ambiguous:

> [1, 2, 3] in ([1, 2, 3])

The ambiguity resolution rule of the language would resolve the above as:

> [1, 2, 3] in <expression> 

Where <expression> is the list [1, 2, 3]. So your conclusion is correct and the result will be false. This is quite confusing to users actually, and deserves a ticket to the RTF.

Reply:
May I ask you to expand further on:

> The following expression is not ambiguous, because the comma in "( <value> , <value> )" disambiguates it.

And 

> the ambiguity resolution rule

?

The only references to ambiguity I know of in the spec relate to names, not unary tests.

Regarding this example would `1 in ([1,2,3,4], [1,2,3])` test for 'in' and not '=' against the lists?  I understood when a list is an element of a unary test we are testing for inclusion in this case. Or both?

Reply:
> The following expression is not ambiguous, because the comma in "( , )" disambiguates it.

Thinking about it, I think you are saying it defines it as a unaryTests and not an 'expression' .. got it.  

But the 'test for inclusion vs equality' thing I'm still not 100% sure on (hence the test).

Reply:
@tarilabs. test updated as such.